body_test.go

```diff
@@ -0,0 +1,166 @@
+package main
```

Comment:
Note, since you're testing a library, rather than a command line utility, it would be useful to give the tests a different package name (such as `test`). This will force the tests to refer to the fully qualified name of the go types they interact with (e.g.

```
_, err := artifact.NewBody("file", 128, 128)
```

The advantage of this is that your test will be calling the library in the same way that a user would call it, and you can be sure that your library has exported the types that are needed by the user.

Note when you move it out of `main` into e.g. `artifact` you can also generate the go docs, and you'll see there which functions/methods/variables are visible externally, which is also useful for seeing how your API is presented to users of it.

Reply:
My goal is to not have newBody be an exposed function.  It's really just an implementation detail.  Given that, would you still recommend doing it this way?

Reply:
Sorry I had wrongly assumed calling code would call this. Indeed, if `newBody` is only called from code inside the package, i.e. calling code calls other functions/methods and doesn't see `newBody`, then yes, not exporting it would be the right choice (i.e. keeping it named `newBody`).